pkg/minikube/tunnel/route_linux.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2018 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tunnel
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"net"
+	"os/exec"
+	"strings"
+)
+
+func (router *osRouter) EnsureRouteIsAdded(route *Route) error {
+	exists, e := isValidToAddOrDelete(router, route)
+	if e != nil {
+		return e
+	}
+	if exists {
+		return nil
+	}
+
+	serviceCIDR := route.DestCIDR.String()
+	gatewayIP := route.Gateway.String()
+
+	glog.Infof("Adding Route for CIDR %s to gateway %s", serviceCIDR, gatewayIP)
+	command := exec.Command("sudo", "ip", "route", "add", serviceCIDR, "via", gatewayIP)
+	glog.Infof("About to run command: %s", command.Args)
+	stdInAndOut, e := command.CombinedOutput()
+	message := fmt.Sprintf("%s", stdInAndOut)
+	if len(message) > 0 {
+		return fmt.Errorf("error adding Route: %s, %d", message, len(strings.Split(message, "\n")))
+	}
+	glog.Infof("%s", stdInAndOut)
+	if e != nil {
+		glog.Errorf("error adding Route: %s, %d", message, len(strings.Split(message, "\n")))
+		return e
+	}
+	return nil
+}
+
+func (router *osRouter) Inspect(route *Route) (exists bool, conflict string, overlaps []string, err error) {
+	command := exec.Command("netstat", "-nr", "-f", "inet")
+	stdInAndOut, e := command.CombinedOutput()
+	if e != nil {
+		err = fmt.Errorf("error running '%v': %s", command, e)
+		return
+	}
+	routeTableString := fmt.Sprintf("%s", stdInAndOut)
```

Comment:
Sprintf is a slow way to convert bytes to strings. Just use string(), but only if you must. I'd argue that parseTable should probably take a slice of bytes.